### DISCUSSION TITLE
Reload extension without window reload in web build

### DISCUSSION THREAD

**Discussion by progrium** [microsoft/vscode-discussions#2389] [Extension Development QnA] [answered]
I'd like to be able to reload just my extension programmatically. However, I'll take reloading the whole extension host. I just don't want to reload the window. Unfortunately, when building for the web, I don't see the command "Developer: Restart Extension Host" ... is there a way to enable it to be called by code? Is there any other way to get VSCode to reload the source for my extension without reloading the window?

**Comment by isidorn** [comment, marked as answer]
VS Code can not just restart your extension. It can restart the whole extension host that has all the extensions running.
You can trigger this functionality from your extension by using the `executeCommand` API.

**Reply by progrium** [reply]
So this command *is* available in the web build I just have to call it via API?

**Reply by isidorn** [reply]
yeah
